File: 13_Roman_to_integer/Solution.java
class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> valores = new HashMap<>();
        valores.put('I', 1);
        valores.put('V', 5);
        valores.put('X', 10);
        valores.put('L', 50);
        valores.put('C', 100);
        valores.put('D', 500);
        valores.put('M', 1000);
        
        int resultado = 0;
        
        for (int i = 0; i < s.length(); i++) {
            int atual = valores.get(s.charAt(i));
            if (i < s.length() - 1 && valores.get(s.charAt(i + 1)) > atual) {
                resultado -= atual;
            } else {
                resultado += atual;
            }
        }
        
        return resultado;
    }
}

File: 13_Roman_to_integer/Solution.py
class Solution(object):
    def romanToInt(self, s):
        valores = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        resultado = 0
        n = len(s)
        for i in range(n):
            valor_atual = valores[s[i]]
            if i < n - 1 and valores[s[i+1]] > valor_atual:
                resultado -= valor_atual
            else:
                resultado += valor_atual
                
        return resultado